crates/sats/src/db/raw_def.rs

```diff
@@ -0,0 +1,336 @@
+        table.product_type_ref = product_type_ref;
+        self.with_table(table)
+    }
+}
+
+/// Represents a sequence definition for a database table column.
+#[derive(Debug, Clone, ser::Serialize, de::Deserialize)]
+#[sats(crate = crate)]
+pub struct RawSequenceDef {
+    /// The name of the column associated with this sequence.
+    pub column_name: Box<str>,
+    pub start: Option<i128>,
+    pub min_value: Option<i128>,
+    pub max_value: Option<i128>,
+}
+
+impl RawSequenceDef {
+    /// Creates a new [RawSequenceDef] instance for a specific table and column.
+    ///
+    /// # Arguments
+    ///
+    /// * `column` - The name of the column in the `table`.
+    pub fn for_column(column: &str) -> Self {
+        //removes the auto-generated suffix...
+
+        RawSequenceDef {
+            column_name: column.into(),
+            start: None,
+            min_value: None,
+            max_value: None,
+        }
+    }
+}
+
+#[derive(Debug, Copy, Clone, Eq, PartialEq, PartialOrd, Ord, Display, Hash, de::Deserialize, ser::Serialize)]
+#[sats(crate = crate)]
+pub enum IndexType {
+    BTree = 0,
+    Hash = 1,
+}
+
+impl From<IndexType> for u8 {
+    fn from(value: IndexType) -> Self {
+        value as u8
+    }
+}
+
+impl TryFrom<u8> for IndexType {
+    type Error = ();
+    fn try_from(v: u8) -> Result<Self, Self::Error> {
+        match v {
+            0 => Ok(IndexType::BTree),
+            1 => Ok(IndexType::Hash),
+            _ => Err(()),
+        }
+    }
+}
+
+/// A struct representing the definition of a database index.
+#[derive(Debug, Clone, ser::Serialize, de::Deserialize)]
+#[sats(crate = crate)]
+pub struct RawIndexDef {
+    /// The type of the index.
+    pub index_type: IndexType,
+    /// List of column names that compose the index.
+    /// ORDER MATTERS HERE.
+    pub column_names: Vec<Box<str>>,
+}
+
+/// A struct representing the definition of a database column.
+#[derive(Debug, Clone, ser::Serialize, de::Deserialize)]
+#[sats(crate = crate)]
+pub struct RawColumnDef {
+    /// The name of the column.
+    pub col_name: Box<str>,
+
+    /// The type of the column.
+    pub col_type: AlgebraicType,
+}
+
+impl RawColumnDef {
+    /// Creates a new [RawColumnDef] for a field with the specified data type.
+    ///
+    /// This method is typically used to define columns with predefined names and data types for tests.
+    ///
+    /// # Arguments
+    ///
+    /// * `field_name`: The name for which to create a column definition.
+    /// * `col_type`: The [AlgebraicType] of the column.
+    ///
+    pub fn new(field_name: &str, col_type: AlgebraicType) -> Self {
+        Self {
+            col_name: field_name.into(),
+            col_type,
+        }
+    }
+}
+
+/// Requires that the projection of the table onto these columns is an bijection.
+#[derive(Debug, Clone, ser::Serialize, de::Deserialize)]
+#[sats(crate = crate)]
+pub struct RawUniqueConstraintDef {
+    pub column_names: UnsortedVec<Box<str>>,
+}
```

Comment:
Ah yes, I was generally moving to string identifiers because I thought I was going to be reordering tables, but it turned out that I'm not doing that here.

I'm still avoiding using `ColList` in `Def`s due to the outstanding issue with orderings in the packed version of `ColList`. E.g. if `IndexDef` had a `ColList` you would lose ordering information. That's fine in this case though.